lib/internal/modules/esm/module_map.js

```diff
@@ -10,25 +12,35 @@ let debug = require('internal/util/debuglog').debuglog('esm', (fn) => {
 const { ERR_INVALID_ARG_TYPE } = require('internal/errors').codes;
 const { validateString } = require('internal/validators');
 
+const validateAssertType = (type) =>
+  type === kImplicitAssertType || validateString(type, 'type');
+
 // Tracks the state of the loader-level module cache
 class ModuleMap extends SafeMap {
   constructor(i) { super(i); } // eslint-disable-line no-useless-constructor
-  get(url) {
+  get(url, type = kImplicitAssertType) {
     validateString(url, 'url');
-    return super.get(url);
+    validateAssertType(type);
+    return super.get(url)?.[type];
   }
-  set(url, job) {
+  set(url, type = kImplicitAssertType, job) {
     validateString(url, 'url');
+    validateAssertType(type);
     if (job instanceof ModuleJob !== true &&
         typeof job !== 'function') {
       throw new ERR_INVALID_ARG_TYPE('job', 'ModuleJob', job);
     }
-    debug(`Storing ${url} in ModuleMap`);
-    return super.set(url, job);
+    debug(`Storing ${url} (${
+      type === kImplicitAssertType ? 'implicit type' : type
+    }) in ModuleMap`);
+    const cachedJobsForUrl = super.get(url) || ObjectCreate(null);
```

Comment:
I mean . . . sure, but why is this an improvement? We know that `super.get` will always return either an object or `undefined`, so `??` is acceptable here, but why is it preferable?

Reply:
after all, it's javascript. without any types, "we" might know after digging thru the code to be sure, "you" might know because you wrote it. 😉  `nullish coalescing` makes your intention for anyone reading the code at a later point in time more clear, otherwise, a reader might as well think: "the syntax has either not existed", "was unknown to the contributor (or simply ignored)", or we might as well deal with falsy values and it was done on purpose.